### PR TITLE
[SPARK-37896][SQL][FOLLOWUP] Fix NPE in ConstantColumnVector.close()

### PR DESCRIPTION
**Pull request by c21**
### What changes were proposed in this pull request?

This PR is a followup of https://github.com/apache/spark/pull/35068 to fix the null pointer exception when calling `ConstantColumnVector.close()`. `ConstantColumnVector.childData` can be null for e.g. non-struct data type.

### Why are the changes needed?

Fix the exception when cleaning up column vector.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Modified unit test in `ConstantColumnVectorSuite.scala` to exercise the code path of `ConstantColumnVector.close()` for every tested data type. Without the fix, the unit test throws NPE.

### CODE REVIEW

File: sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ConstantColumnVector.java [hunk 1]
```diff
@@ -70,10 +70,15 @@ public ConstantColumnVector(int numRows, DataType type) {
 
   @Override
   public void close() {
+    stringData = null;
```
Comment (c21) [inline sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ConstantColumnVector.java:73]: This change is not related to the NPE fix, but it's good to have. We should release the memory for `UTF8String` as well.

**Comment by c21** [comment]
cc @cloud-fan and @Yaohua628

**Comment by Yaohua628** [comment]
thanks for the fix!

**Review by Yaohua628 — APPROVED** [review]
LGTM, thanks!

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by c21** [comment]
Thank you @cloud-fan and @Yaohua628 for review!

**Pull request closed without merging**